code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -589,3 +589,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			user << "<span class='warning'>You need to dry this first!</span>"
 	else
 		..()
+///////////////////
+//Strike anywhere//
+///////////////////
+/obj/item/clothing/mask/cigarette/strikeanywhere
+	name = "strike anywhere cigarette"
+	desc = "A cigarette infused with a proprietary compound that ignites when struck. Smells of magnesium."
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/afterattack(atom/target, mob/user, proximity)
+	if(!lit && smoketime > 0)
+		user.visible_message("[user] strikes [target] with the [src] and it ignites!", "<span class='notice'>You strike [target] with the [src] and it ignites!</span>")
+		light()
+	..()
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/throw_impact(atom/hit_atom)
+	if(!lit && smoketime > 0)
+		hit_atom.visible_message("The [src] strikes [hit_atom] and ignites!")
+		light()
+	..()

```

Comment:
I'm not sure but I think changing your throw_impact() to
```
. = ..()
if(.)
	your stuff here
```
might work. atom/proc/throw_impact() is "return A.hitby() and hitby() returns 1 if we actually hit the atom (this is how we handle mobs catching thrown items, it returns 0 in that case)